src/utils/dom.ts

```diff
@@ -78,7 +78,16 @@ export function queryElementsRoots<T extends Element = Element>(element: Element
 
 // Queries an element's rootNode and any ancestor rootNodes.
 // based on https://stackoverflow.com/q/54520554/194216
-export function queryElementRoots<T extends Element = Element>(element: Element, selector: string): T | null {
+export function queryElementRoots<T extends Element = Element>(
+  element: Element,
+  {
+    selector,
+    id
+  }: {
+    selector?: string;
```

Comment:
Can you doc that `id` takes precedence if both are specified? It might not be clear from the signature alone.